src/data/sigma/order.lean

```diff
@@ -181,5 +195,42 @@ instance bounded_order [partial_order ι] [bounded_order ι] [Π i, preorder (α
   bounded_order (Σₗ i, α i) :=
 { .. lex.order_bot, .. lex.order_top }
 
+instance [preorder ι] [densely_ordered ι] [Π i, nonempty (α i)] [Π i, preorder (α i)]
+  [Π i, densely_ordered (α i)] :
+  densely_ordered (Σₗ i, α i) :=
+⟨begin
+  simp_rw sigma.lex.lt_def,
+  rintro ⟨i, a⟩ ⟨j, b⟩ (h | ⟨(rfl : i = j), h⟩),
+  { obtain ⟨k, hi, hj⟩ := exists_between h,
+    obtain ⟨c⟩ : nonempty (α k) := infer_instance,
+    exact ⟨⟨k, c⟩, or.inl hi, or.inl hj⟩ },
+  { obtain ⟨c, ha, hb⟩ := exists_between h,
+    exact ⟨⟨i, c⟩, or.inr ⟨rfl, ha⟩, or.inr ⟨rfl, hb⟩⟩ }
```

Comment:
and this could be
```suggestion
  rintro ⟨i, a⟩ ⟨j, b⟩ (⟨_, _, h⟩ | ⟨_, b, h⟩),
  { obtain ⟨c, ha⟩ := exists_gt a,
    exact ⟨⟨i, c⟩, right _ _ ha, left _ _ h⟩ },
  { obtain ⟨c, ha, hb⟩ := exists_between h,
    exact ⟨⟨i, c⟩, right _ _ ha, right _ _ hb⟩ }
```
(btw `sigma.lex.` can be omitted from `lt_def`)